scripts/west_commands/run_common.py

```diff
@@ -165,21 +195,101 @@ def do_run_common(command, user_args, user_runner_args, domains=None):
             # Get the user specified domains.
             domains = load_domains(build_dir).get_domains(user_args.domain)
 
-    if len(domains) > 1 and len(user_runner_args) > 0:
-        log.wrn("Specifying runner options for multiple domains is experimental.\n"
-                "If problems are experienced, please specify a single domain "
-                "using '--domain <domain>'")
+    if len(domains) > 1:
+        if len(user_runner_args) > 0:
+            log.wrn("Specifying runner options for multiple domains is experimental.\n"
+                    "If problems are experienced, please specify a single domain "
+                    "using '--domain <domain>'")
+
+        # Process all domains to load board names and populate flash runner
+        # parameters.
+        board_names = set()
+        for d in domains:
+            if d.build_dir is None:
+                build_dir = get_build_dir(user_args)
+            else:
+                build_dir = d.build_dir
+
+            cache = load_cmake_cache(build_dir, user_args)
+            board = cache['CACHED_BOARD']
```

Comment:
Just remembered that `CACHED_BOARD` could be abbreviated like `nrf5340dk//cpunet`, where the regexes won't work.
Although you have to deliberately put e.g. `-Dremote_BOARD=nrf5340dk//cpunet` and run `--recover` to notice this.
Still, maybe `CONFIG_BOARD_TARGET` is needed here instead.